packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Snippets/Snippet.New.cs

```diff
@@ -16,8 +16,8 @@ public static partial class Snippet
     {
         public static class New
         {
-            public static ValueExpression ArgumentOutOfRangeException(EnumProvider enumType, ParameterProvider valueParameter)
-                => Instance(typeof(ArgumentOutOfRangeException), Nameof(valueParameter), valueParameter, Literal($"Unknown {enumType.Name} value."));
+            public static ValueExpression ArgumentOutOfRangeException(TypeProvider provider, ParameterProvider valueParameter)
+                => Instance(typeof(ArgumentOutOfRangeException), Nameof(valueParameter), valueParameter, Literal($"Unknown {provider.Name} value."));
```

Comment:
well changing it to `TypeProvider` makes it to make less sense right now.
We do not know if this typeprovider is an enum therefore this has the possibility of creating an exception which makes no sense. I do not quite like this.

Reply:
added a debug assert here.  In order to call this outside of mgc we need this method to take in a typeprovider.